### ISSUE TITLE
Add environment variable plugin

### ISSUE THREAD

**Issue by jafreck** [Azure/aztk#544]

**Comment by timotheeguerin** [comment]
should that be a plugin though?

**Comment by jafreck** [comment]
I feel like it would be a very easy interface to specify environment variables. Potentially this could be a top level feature in cluster.yaml though. I would be fine with it either way, plugins just seem like a easy fit to implement, but I don't think it's necessary for the feature.

**Comment by timotheeguerin** [comment]
Yeah I believe it would nicer as a global setting as this could be used in plugins too and I would expect it to be set before them

**Comment by jafreck** [comment]
Yeah I like that idea much better

**Comment by jafreck** [comment]
edited to reflect @timotheeguerin suggestion
